### DISCUSSION TITLE
POLL: What Hugo version are you on?

### DISCUSSION THREAD

**Discussion by kaushalmodi** [kaushalmodi/ox-hugo#481] [General]
Hello, 

I plan to do a cleanup of the ox-hugo package and move out the Blackfriday related code to a separate library that I will stop maintaining.

Hugo switched to using the Go Commonmark parsing library Goldmark as the default Markdown parser [about 5 months ago with v0.60.0](https://gohugo.io/news/0.60.0-relnotes/).

Given that that switch go Commonmark library happened a while ago, I am planning to stop maintaining the `ox-blackfriday.el` and other Blackfriday related code in `ox-hugo.el` by move those pieces out of `ox-hugo.el`.

Before I make that switch, I want to gauge how many current ox-hugo users have upgraded to Hugo v0.60.0 or newer.

## Poll

_Click on the respective emoji here to put your vote._

![image](https://user-images.githubusercontent.com/3578197/80385293-e30b3300-8873-11ea-9791-a0527c65a547.png)

**I use ox-hugo and ..**

- 🚀 - I am using Hugo 0.60.0 or newer
- 👍 - I am still using Hugo 0.59.1 or older Hugo, and I plan to upgrade
- 👎 - I am still using Hugo 0.59.1 or older Hugo, and I do not plan to upgrade

**Comment by titaniumbones** [comment]
I will do whatever you recommend :-). I'm having trouble identifying black Friday features that are not supported in Goldmark - thebdocs mention the existence of such features several times, but they don't enumerate them. Do you have any insight?

On Apr. 27, 2020 10:15 a.m., Kaushal Modi <notifications@github.com> wrote:

/polls "I am using Hugo 0.60.0 or newer" "I am still using Hugo 0.59.1 or older Hugo, and I plan to upgrade" "I am still using Hugo 0.59.1 or older Hugo, and I do not plan to upgrade"

**Comment by kaushalmodi** [comment]
> I will do whatever you recommend :-).

I strongly believe in upgrading the newer and spec-compliant Commonmark based Goldmark library. It's a lot less headache from the aspect of ox-hugo.el maintenance and users also see a more consistent behavior in how the Markdown gets parsed. 

As a bonus, Commonmark also has this online Markdown parser to see what syntax works and doesn't: https://spec.commonmark.org/dingus/ (and you can almost bet that the same will be parsed by Goldmark -- that Hugo now defaults to).

So, yes, if you haven't already, upgrade to the latest stable version of Hugo.

> I'm having trouble identifying black Friday features that are not supported in Goldmark 

I would suggest posting on https://discourse.gohugo.io/ about the Blackfriday feature you liked, but didn't get into Goldmark.

---

> Do you have any insight?

The Hugo docs list which extensions or features work for which parser.

- [Goldmark extensions][1]
- [Blackfriday extensions][2]

[1]: https://gohugo.io/getting-started/configuration-markup/#goldmark
[2]: https://gohugo.io/getting-started/configuration-markup/#blackfriday

**Comment by titaniumbones** [comment]
I think I'll be fine, after looking through the main issues here. I am not quite certain what the current status of this package is, though -- are we currently just exporting to commonmark and bypassing ox-blackfriday, or is blackfriday MD close enough so that most stuff works with ox-blackfriday even if the website is using goldmark?

**Comment by kaushalmodi** [comment]
> I am not quite certain what the current status of this package is, though -- are we currently just exporting to commonmark and bypassing ox-blackfriday, or is blackfriday MD close enough so that most stuff works with ox-blackfriday even if the website is using goldmark?

Actually all the normal Markdown stuff that worked with Blackfriday with the Blackfriday-specific hacks added by ox-blackfriday, work for Goldmark as well! (Note that when switch to Hugo v0.60.0+ that uses Goldmark, you need this extra setting in your hugo sites' `config.toml` files: https://github.com/kaushalmodi/ox-hugo/issues/307 .) Those hacks don't hurt the Goldmark parsing.

_Once I have a separate ox-commonmark.el, I can remove those hacks and the Goldmark parsing will still work._

**Comment by emacsomancer** [comment]
My issue is that I use the Academic theme on one site, and v.3.0 of Academic doesn't work with Hugo 0.60.0+ and it's really complicated to upgrade versions of Academic (I would have to hand-upgrade through at least 11 versions each with its own set of breaking changes to get to a Hugo 0.60.0 version of Academic).

**Comment by kaushalmodi** [comment]
@emacsomancer This cleanup **will not** make `ox-hugo` stop supporting Blackfriday. I will move ox-blackfriday related hacks to its own library and have the Markdown for Goldmark get generated from a cleaner version of ox-blackfriday. 

If your site is working fine with ox-hugo right now with the version of hugo binary you have, that will still stay the same.

**Comment by kaushalmodi** [comment]
Today I took the first step towards gradually removing the Blackfriday related hacks. Please see/subscribe this announcement: https://github.com/kaushalmodi/ox-hugo/discussions/485
